lib/cylc/network/https/base_client.py

```diff
@@ -138,28 +139,21 @@ def call_server_func(self, category, *func_dicts, **fargs):
         else:
             func_dict = None
 
-        if self.host is None and self.port is not None:
-            self.host = get_hostname()
         try:
             self._load_contact_info()
         except (IOError, ValueError, SuiteServiceFileError):
             raise ConnectionInfoError(self.suite)
         handle_proxies()
-        host = self.host
-        if host == "localhost":
-            host = get_hostname().split(".")[0]
```

Comment:
Interestingly, due to some DNS changes (presumably) this line recently broke `cylc scan` on my desktop box at work (where hostname `foo.niwa.local` is visible, but `foo` isn't).